starknet_py/proxy/proxy_check.py

```diff
@@ -77,3 +77,25 @@ async def _get_storage_at_or_none(
             block_hash="latest",
         )
         return result or None
+
+
+class EthProxyCheck(ProxyCheck):
+    async def implementation_address(
+        self, address: Address, client: Client
+    ) -> Optional[int]:
+        return await self.implementation(address, client)
+
+    async def implementation_hash(
+        self, address: Address, client: Client
+    ) -> Optional[int]:
+        return await self.implementation(address, client)
```

Comment:
I think we can verify if it returns implementation address or class hash and explicitly support only one of them (return None in the other case).